From 38497b12567b25cb328f5ca316db771fcf67ebfd Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 25 Feb 2015 14:35:10 -0800
Subject: [PATCH 0149/1851] dt-bindings: dma: xilinx: dpdma: DT bindings for
 Xilinx DPDMA

commit 6beabbb00f8096b4e68273acb2b4ac641aeb9fda from
https://github.com/Xilinx/linux-xlnx.git

The ZynqMP includes the DisplayPort subsystem with its own DMA engine
called DPDMA. The DPDMA IP comes with 6 individual channels
(4 for display, 2 for audio). This documentation describes DT bindings
of DPDMA.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: waiting (dmaengine)
Link: https://lore.kernel.org/dmaengine/20191107021400.16474-2-laurent.pinchart@ideasonboard.com/
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/dma/xilinx/xilinx_dpdma.txt      | 91 +++++++++++++++++++
 1 file changed, 91 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/dma/xilinx/xilinx_dpdma.txt

diff --git a/Documentation/devicetree/bindings/dma/xilinx/xilinx_dpdma.txt b/Documentation/devicetree/bindings/dma/xilinx/xilinx_dpdma.txt
new file mode 100644
index 000000000000..5f1e680ffcc2
--- /dev/null
+++ b/Documentation/devicetree/bindings/dma/xilinx/xilinx_dpdma.txt
@@ -0,0 +1,91 @@
+Device-Tree bindings for Xilinx ZynqMP DisplayPort Subsystem
+
+The ZynqMP DisplayPort subsystem handles DMA channel buffer management,
+blending, and audio mixing. The DisplayPort subsystem receives display
+and audio frames from DPDMA and transmits output to the DisplayPort IP core.
+
+Required properties:
+ - compatible: Should be "xlnx,dpdma".
+ - reg: Base address and size of the IP core.
+ - interrupts: Interrupt number.
+ - interrupts-parent: phandle for interrupt controller.
+ - clocks: phandle for AXI clock
+ - clock-names: The identification string, "axi_clk", is always required.
+
+Required child node properties:
+- compatible: Should be one of "xlnx,video0", "xlnx,video1", "xlnx,video2",
+  "xlnx,graphics", "xlnx,audio0", or "xlnx,audio1".
+
+Example:
+
+	xlnx_dpdma: axidpdma@43c10000 {
+		compatible = "xlnx,dpdma";
+		reg = <0x43c10000 0x1000>;
+		interrupts = <0 54 4>;
+		interrupt-parent = <&intc>;
+		clocks = <&clkc 16>;
+		clock-names = "axi_clk";
+		xlnx,axi-clock-freq = <200000000>;
+
+		dma-channels = <6>;
+
+		#dma-cells = <1>;
+		dma-video0channel@43c10000 {
+			compatible = "xlnx,video0";
+		};
+		dma-video1channel@43c10000 {
+			compatible = "xlnx,video1";
+		};
+		dma-video2channel@43c10000 {
+			compatible = "xlnx,video2";
+		};
+		dma-graphicschannel@43c10000 {
+			compatible = "xlnx,graphics";
+		};
+		dma-audio0channel@43c10000 {
+			compatible = "xlnx,audio0";
+		};
+		dma-audio1channel@43c10000 {
+			compatible = "xlnx,audio1";
+		};
+	};
+
+* DMA client
+
+Required properties:
+- dmas: a list of <[DPDMA device phandle] [Channel ID]> pairs. "Channel ID"
+  is defined as video0 = 0, video1 = 1, video2 = 2, graphics = 3, audio0 = 4,
+  and audio1 = 5.
+
+Example:
+
+	xilinx_drm {
+		compatible = "xlnx,drm";
+		xlnx,encoder-slave = <&xlnx_dp>;
+		clocks = <&si570 0>;
+		xlnx,connector-type = "DisplayPort";
+		xlnx,dp-sub = <&xlnx_dp_sub>;
+		planes {
+			xlnx,pixel-format = "rgb565";
+			plane0 {
+				dmas = <&xlnx_dpdma 3>;
+				dma-names = "dma";
+			};
+			plane1 {
+				dmas = <&xlnx_dpdma 0>;
+				dma-names = "dma";
+			};
+		};
+	};
+
+	xlnx_dp_snd_pcm0: dp_snd_pcm0 {
+		compatible = "xlnx,dp-snd-pcm";
+		dmas = <&xlnx_dpdma 4>;
+		dma-names = "tx";
+	};
+
+	xlnx_dp_snd_pcm1: dp_snd_pcm1 {
+		compatible = "xlnx,dp-snd-pcm";
+		dmas = <&xlnx_dpdma 5>;
+		dma-names = "tx";
+	};
-- 
2.31.1

